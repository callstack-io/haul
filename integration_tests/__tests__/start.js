/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

const { runHaul, runHaulSync } = require('../runHaul');
const { cleanup, writeFiles, replaceTestPath } = require('../utils');
const path = require('path');
const os = require('os');
const { run } = require('../utils');
const stripAnsi = require('strip-ansi');

const TEMP_DIR = path.resolve(os.tmpdir(), 'start_test');
const TEST_PROJECT_DIR = path.resolve(
  __dirname,
  '../fixtures/react-native-generated-project',
);

beforeAll(() => {
  run('yarn', TEST_PROJECT_DIR);
});

beforeEach(() => cleanup(TEMP_DIR));
afterEach(() => cleanup(TEMP_DIR));

test('start command displays "Select platform" message', () => {
  writeFiles(TEMP_DIR, {
    'webpack.haul.js': '{}',
  });

  const { stdout } = runHaulSync(TEMP_DIR, ['start']);
  expect(stdout).toMatchSnapshot();
});

test('start --platform ios', done => {
  testPlatform('ios', done);
});

test('start --platform android', done => {
  testPlatform('android', done);
});

function testPlatform(platform, done) {
  // $FlowFixMe â€“ new Jest API not yet in typings
  jest.setTimeout(20000);
  expect.assertions(1);
  const messageBuffer = [];
  const haul = runHaul(TEST_PROJECT_DIR, ['start', '--platform', platform]);

  haul.stdout.on('data', data => {
    const message = stripAnsi(data.toString()).trim();
    if (message.length > 0) {
      messageBuffer.push(message);
    }

    if (message.match('Built successfully!')) {
      expect(replaceTestPath(messageBuffer.join('\n'))).toMatchSnapshot();
      done();
      haul.kill();
    }
  });
}
