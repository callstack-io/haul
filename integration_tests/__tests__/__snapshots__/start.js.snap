// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start --platform android 1`] = `
"WARN  Error running adb reverse tcp:8081 tcp:8081: error: device '(null)' not found
error: device '(null)' not found
INFO  Ready at http://localhost:8081

Haul is now bundling your React Native app, starting from:
<<REPLACED>>/fixtures/react-native-generated-project/index.android.js

A fresh build may take longer than usual
DONE  Built successfully!

You can now run the app on your android device"
`;

exports[`start --platform ios 1`] = `
"INFO  Ready at http://localhost:8081

Haul is now bundling your React Native app, starting from:
<<REPLACED>>/fixtures/react-native-generated-project/index.ios.js

A fresh build may take longer than usual
DONE  Built successfully!

You can now run the app on your ios device"
`;

exports[`start command displays "Select platform" message 1`] = `
"? Select platform to bundle for (Use arrow keys)
‚ùØ ios - Serves iOS bundle 
  android - Serves Android bundle 
  all - Serves both platforms [30D[30C
"
`;
